fix: non-authority player instance spawns with no motion

NoelStephensUnity opened the pull request:
### Description
Fixes issue where a non-authority player instance could improperly initialize when two clients are connecting to the same session at the "relative" same time. Removing spam logging on collision messages.

### Issue Number(s)
Based on [NGO user issue #3412 ](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/issues/3412#issuecomment-2815726466)

### Contribution checklist
 - [ ] Tests have been added for the project and/or any internal package
 - [ ] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [ ] JIRA ticket ID is in the PR title or at least one commit message
 - [ ] Include the ticket ID number within the body message of the PR to create a hyperlink

RikuTheFuffs commented:
I moved the target to develop as requested by @fernando-cortez  . @NoelStephensUnity could you please resync the branch, and add a changelog line so we can merge this?

NoelStephensUnity commented:
> I moved the target to develop as requested by @fernando-cortez . @NoelStephensUnity could you please resync the branch, and add a changelog line so we can merge this?

- develop branch has been merged. 👍 
- added two change log entries. 👍 

@RikuTheFuffs Let me know if there is anything else needed for this PR? 
_(I also validated the the fix after having merged the develop branch)_

fernando-cortez commented:
> @fernando-cortez Not sure why the above tests are failing... as they are failing in other areas of the repository.

I'll take a look!
